File: src/main/java/ru/mm/surv/capture/service/impl/FfmpegLogger.java
package ru.mm.surv.capture.service.impl;

import lombok.extern.slf4j.Slf4j;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

@Slf4j
class FfmpegLogger implements Runnable {

    private final String streamName;
    private final BufferedReader reader;

    public FfmpegLogger(String streamName, InputStream stream) {
        this.streamName = streamName;
        this.reader = new BufferedReader(new InputStreamReader(stream));
    }

    @Override
    public void run() {
        try {
            String line;
            while ((line = reader.readLine()) != null) {
                log.info("{} {}", streamName, line);
            }
        } catch (IOException e) {
            log.error(streamName + " " + e.getMessage(), e);
        }
    }
}

File: backend/src/main/kotlin/ru/mm/surv/capture/service/impl/FfmpegLogger.kt
package ru.mm.surv.capture.service.impl

import lombok.extern.slf4j.Slf4j
import org.slf4j.LoggerFactory
import java.lang.Runnable
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader

@Slf4j
internal class FfmpegLogger(private val streamName: String, stream: InputStream) : Runnable {

    private val log = LoggerFactory.getLogger(FfmpegLogger::class.java)

    private val reader = BufferedReader(InputStreamReader(stream))

    override fun run() {
        try {
            var line: String?
            while (reader.readLine().also { line = it } != null) {
                log.info("{} {}", streamName, line)
            }
        } catch (e: IOException) {
            val errorMessage = e.message
            log.error("$streamName $errorMessage", e)
        }
    }
}
